crates/hyperdrive-math/src/long/fees.rs

```diff
@@ -5,36 +5,54 @@ use fixed_point_macros::fixed;
 use crate::State;
 
 impl State {
-    /// Calculates the curve fee paid by longs for a given base amount.
+    /// Calculates the curve fee paid when opening longs with a given base amount.
     ///
-    /// The curve fee $c(x)$ paid by longs is paid in bonds and is given by:
+    /// The open long curve fee, $\Phi_{c,ol}(\Delta x)$, is paid in bonds and
+    /// is given by:
     ///
     /// $$
-    /// c(x) = \phi_c \cdot \left( \tfrac{1}{p} - 1 \right) \cdot x
+    /// \Phi_{c,ol}(\Delta x) = \phi_c \cdot \left( \tfrac{1}{p} - 1 \right) \cdot \Delta x
     /// $$
-    pub fn open_long_curve_fees(&self, base_amount: FixedPoint) -> FixedPoint {
-        self.curve_fee()
-            * ((fixed!(1e18) / self.calculate_spot_price()) - fixed!(1e18))
-            * base_amount
+    pub fn open_long_curve_fee(&self, base_amount: FixedPoint) -> FixedPoint {
+        // NOTE: Round up to overestimate the curve fee.
+        (fixed!(1e18).div_up(self.calculate_spot_price()) - fixed!(1e18))
+            .mul_up(self.curve_fee())
+            .mul_up(base_amount)
```

Comment:
How does this work? And the units returned is in bonds?

Looking at this: https://github.com/delvtech/hyperdrive/blob/33982bba72d861b3b90cb0b4083fec30accb649f/contracts/src/internal/HyperdriveBase.sol#L773-L776

I would think we'd have to do something like this to get the curve fee in bonds:

```
curve_fee
  .mul_up(1 - spot_price)
  .mul_up(base_amount.div_up(spot_price))
  .mul_up(normalize_time_remaining)
```

Reply:
For open long we use this: https://github.com/delvtech/hyperdrive/blob/33982bba72d861b3b90cb0b4083fec30accb649f/contracts/src/internal/HyperdriveBase.sol#L718